src/Experiment/src/CustomAttributeBuilder.cs

```diff
@@ -0,0 +1,581 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers.Binary;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+
+namespace System.Reflection.Emit.Experimental
+{
+    public class CustomAttributeBuilder
+    {
+        internal ConstructorInfo m_con;
+        private object?[] m_constructorArgs;
+        internal byte[] m_blob;
```

Comment:
NIT: it would be good if we can also add some comments that explain what is this used for, and why it's internal instead of private.